pkg/scaling/executor/scale_scaledobjects.go

```diff
@@ -216,9 +216,15 @@ func (e *scaleExecutor) scaleToZeroOrIdle(ctx context.Context, logger logr.Logge
 		cooldownPeriod = time.Second * time.Duration(defaultCooldownPeriod)
 	}
 
+	// if scaledObject.Spec.AllowInitScale is not set, then set the default value (true)
+	allowInitScale := true
+	if scaledObject.Spec.AllowInitScale != nil {
+		allowInitScale = *scaledObject.Spec.AllowInitScale
+	}
+
 	// LastActiveTime can be nil if the ScaleTarget was scaled outside of KEDA.
 	// In this case we will ignore the cooldown period and scale it down
-	if scaledObject.Status.LastActiveTime == nil ||
+	if (scaledObject.Status.LastActiveTime == nil && allowInitScale) ||
```

Comment:
I don't think you need the opening and closing parentheses here

Reply:
You should also update the coment above the `if` to cover the new property.